### PR TITLE
Fix 4-state value support for $readmem (#5070)

### PR DESCRIPTION
**Pull request by sifferman**
Fixes https://github.com/verilator/verilator/issues/5070

(I still need to add a couple tests, but I'm posting the PR now in case you have early suggestions).

Changes:

* A `z` or `x` in a data string will be converted to a random bit/nibble according to `$readmemb`/`$readmemh` and `+verilator+rand+reset`
* A 4-state value in an address will now result in a syntax error.

### CODE REVIEW

File: include/verilated.cpp [hunk 1]
```diff
@@ -2032,11 +2032,16 @@ bool VlReadMem::get(QData& addrr, std::string& valuer) {
     while (true) {
         int c = std::fgetc(m_fp);
         if (VL_UNLIKELY(c == EOF)) break;
+        const bool c_is_4state_bin
+            = c == '0' || c == '1' || c == 'x' || c == 'X' || c == 'z' || c == 'Z';
```
Comment (wsnyder) [inline include/verilated.cpp:2036]: Please use Verilator mixed cap style.

```suggestion
        const bool chIs4StateBin
            = c == '0' || c == '1' || c == 'x' || c == 'X' || c == 'z' || c == 'Z';
```

File: include/verilated.cpp [hunk 2]
```diff
@@ -2032,11 +2032,16 @@ bool VlReadMem::get(QData& addrr, std::string& valuer) {
...
+        const bool c_is_2state_hex = std::isxdigit(c);
+        const bool c_is_4state_hex
+            = std::isxdigit(c) || c == 'x' || c == 'X' || c == 'z' || c == 'Z';
```
Comment (wsnyder) [inline include/verilated.cpp:2039]:
```suggestion
            = std::isxdigit(c) || cIs4StateBin;
```

**Review by wsnyder — COMMENTED** [review]
Thanks for working on this, just naming style nit.

File: include/verilated.cpp (hunk 1 above)
Comment (sifferman) [reply include/verilated.cpp:2036]: I can make this change, but I was trying to match the existing naming style in the function. Should What are your thoughts?

Comment (wsnyder) [reply include/verilated.cpp:2036]: Ah, probably I let it slide when the function was first introduced.  For local variables it's not a big deal, either way.

**Comment by sifferman** [comment]
@wsnyder I think this PR is good to go. I renamed all the local variables to mixed cap and added 2 tests

**Pull request merged**